### PR TITLE
fix: Create TTL index for removing expired logs

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

File: store/mongo/migration_1_1_1.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// Up creates indexes for fetching event documents
+func (m *migration_1_1_1) Up(from migrate.Version) error {
+	ctx := context.Background()
+	eventModel := mongo.IndexModel{
+		Keys: bson.D{
+			{Key: KeyEventTs, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(IndexNameTTL).
+			SetExpireAfterSeconds(0),
```
Comment (merlin-northern) [inline store/mongo/migration_1_1_1.go:45]: maybe I am lost, so where do we set after how long the record is removed?

Comment (alfrunes) [reply store/mongo/migration_1_1_1.go:45]: It is [set on insertion](https://github.com/mendersoftware/iot-manager/blob/41a7b3526c4e15d8af226105abf12e7ae59e905a/store/mongo/datastore_mongo_events.go#L84).

**Comment by alfrunes** [comment]
@mender-test-bot start pipeline --fast

File: store/mongo/migration_1_1_1.go (hunk 1 above)
Comment (alfrunes) [reply store/mongo/migration_1_1_1.go:45]: Oops that made me realize, I'm indexing the wrong key! :anguished:
Fixing...

File: store/mongo/migration_1_1_1_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	var foundIndex bool
+	for _, spec := range specs {
+		if spec == nil {
+			continue
+		}
+		if spec.Name == IndexNameTTL {
+			foundIndex = true
+			assert.True(t, spec.ExpireAfterSeconds != nil &&
+				*spec.ExpireAfterSeconds == 0, "TTL property not set")
+		}
```
Comment (merlin-northern) [inline store/mongo/migration_1_1_1_test.go:54]: name is one thing, could we verify that it is the correct field, and also: nit, but we can actually check that index works, by adding some records and sleeping.

**Review by merlin-northern — APPROVED** [review]
one mean comment, looks good.

File: store/mongo/migration_1_1_1_test.go (hunk 2 above)
Comment (alfrunes) [reply store/mongo/migration_1_1_1_test.go]: Added an assertion on the index keys.
For the second point about checking "whether the index works", I'm not sure what you mean, but I believe that is out of the scope of our product.

**Review by merlin-northern — APPROVED** [review]
thank you

**Pull request merged**